solidity/contracts/Gravity.sol

```diff
@@ -252,16 +248,26 @@ contract Gravity is ReentrancyGuard {
 			"Supplied current validators and powers do not match checkpoint."
 		);
 
+		uint256 cumulativePower = 0;
+		for (uint256 i = 0; i < _newPowers.length; i++) {
+			cumulativePower = cumulativePower + _newPowers[i];
+			if (cumulativePower > state_powerThreshold) {
+				break;
+			}
+		}
+		require(
+			cumulativePower > state_powerThreshold,
+			"Submitted validator set signatures do not have enough power."
+		);
+
```

Comment:
This is not useful IMO. You are only protecting the validators from themselves. This logic should be, and is, on the cosmos side.

Reply:
This is also related to the issue https://github.com/PeggyJV/gravity-bridge/issues/257